src/tools/tidy/src/tests_revision_unpaired_stdout_stderr.rs

```diff
@@ -98,14 +98,20 @@ pub fn check(tests_path: impl AsRef<Path>, bad: &mut bool) {
         for sibling in files_under_inspection.iter().filter(|f| {
             f.extension().map(OsStr::to_str).flatten().is_some_and(|ext| EXTENSIONS.contains(&ext))
         }) {
-            let filename_components = sibling.to_str().unwrap().split('.').collect::<Vec<_>>();
-            let file_prefix = filename_components[0];
+            let Some(filename) = sibling.file_name().map(OsStr::to_str).flatten() else {
+                continue;
+            };
+
+            let filename_components = filename.split('.').collect::<Vec<_>>();
+            let [file_prefix, ..] = &filename_components[..] else {
+                continue;
+            };
```

Comment:
I think this logic here isn't robust either. If a test name is `a.b.rs`, and previously we use `file_stem`, the test name becomes `a.b`. But now this sibling check produces `a` which will silent let `a.b.unknown-revision.stdout` go through because there is no entry in `test_info`. I'll have to think about this part of the logic a bit.

Reply:
I'm actually not sure if we *can* properly check here, if we don't assume test name itself to not have any dots.

According to rustc-dev-guide, a test output can take the form

```
<test_name>[.<revision>][.<compare_mode>].<extension>
```

- but `<extension>` can be:
    - `stderr`, `stdout`
    - `run.stderr`, `run.stdout`
    - `64bit.stderr`, `32bit.stderr`
- but `<test_name>` can also have dots if we don't assume it can't have dots.

Maybe it's just better to just say "we assume a test's name don't contain dots or else this tidy check won't catch stray stdout/stderr files for that test"?

Reply:
Yes, I think that's necessary to assume for file names. We can assume that of a file name even if we can't assume that for the full path.

Reply:
> we assume a test's name don't contain dots or else this tidy check won't catch stray stdout/stderr files for that test

We may need to add an assertion with a message to ensure that it doesn't happen.

Reply:
I'll add an aasert for if the test name contains dots, and update the dev-guide to describe this behavior.

Reply:
Probably a good idea. A lot of the test infra assumes that things like e.g. `.64bit.stderr` remain coherent.